File: app/src/main/java/com/thecode/infotify/utils/NetworkChangeReceiver.java
package com.thecode.infotify.utils;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.thecode.infotify.InfotifyApp;


public class NetworkChangeReceiver extends BroadcastReceiver {
    public static ConnectivityReceiverListener connectivityReceiverListener;
    boolean isConnected = false;

    public NetworkChangeReceiver() {
        super();
    }

    @Override
    public void onReceive(Context context, Intent arg1) {

         isConnected = NetworkUtil.getConnectivityStatus(InfotifyApp.getInstance().getApplicationContext()) > 0;
        if (connectivityReceiverListener != null) {
            connectivityReceiverListener.onNetworkConnectionChanged(isConnected);
        }

    }

    public static boolean isConnected() {
        return NetworkUtil.getConnectivityStatus(InfotifyApp.getInstance().getAppContext().getApplicationContext()) > 0;
    }

    public static void connectionNotAvaillable() {

        connectivityReceiverListener.onInternetUnavailable();
    }

    public interface ConnectivityReceiverListener {
        void onNetworkConnectionChanged(boolean isConnected);

        void onInternetUnavailable();

    }
}

File: app/src/main/java/com/thecode/infotify/utils/NetworkChangeReceiver.kt
package com.thecode.infotify.utils

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.thecode.infotify.InfotifyApp

class NetworkChangeReceiver : BroadcastReceiver() {
    var isConnected = false
    override fun onReceive(context: Context, arg1: Intent) {
        isConnected =
            NetworkUtil.getConnectivityStatus(InfotifyApp.instance!!.applicationContext) > 0
        if (connectivityReceiverListener != null) {
            connectivityReceiverListener!!.onNetworkConnectionChanged(
                isConnected
            )
        }
    }

    interface ConnectivityReceiverListener {
        fun onNetworkConnectionChanged(isConnected: Boolean)
        fun onInternetUnavailable()
    }

    companion object {
        var connectivityReceiverListener: ConnectivityReceiverListener? =
            null

        fun isConnected(): Boolean {
            return NetworkUtil.getConnectivityStatus(
                InfotifyApp.instance!!.appContext.applicationContext
            ) > 0
        }

        fun connectionNotAvaillable() {
            connectivityReceiverListener!!.onInternetUnavailable()
        }
    }
}